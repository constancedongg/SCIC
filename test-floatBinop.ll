; ModuleID = 'scic'
source_filename = "scic"

@fmt = private unnamed_addr constant [4 x i8] c"%d\0A\00"
@fmt.1 = private unnamed_addr constant [4 x i8] c"%g\0A\00"
@fmt.2 = private unnamed_addr constant [4 x i8] c"%s\0A\00"

declare i8* @printf(i8*, ...)

declare i32 @printbig(i32)

define i32 @main() {
entry:
  %printf = call i8* (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.1, i32 0, i32 0), double 4.076000e+00)
  %printf1 = call i8* (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.1, i32 0, i32 0), double -1.676000e+00)
  %printf2 = call i8* (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.1, i32 0, i32 0), double 0x400B9C0EBEDFA43F)
  %printf3 = call i8* (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.1, i32 0, i32 0), double 5.000000e-01)
  %printf4 = call i8* (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.1, i32 0, i32 0), double 0x3FDAB8EB281560BC)
  %printf5 = call i8* (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt, i32 0, i32 0), i1 true)
  %printf6 = call i8* (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt, i32 0, i32 0), i1 true)
  %printf7 = call i8* (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt, i32 0, i32 0), i1 true)
  %printf8 = call i8* (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt, i32 0, i32 0), i1 false)
  %pow32 = call double @llvm.pow.f64(double 2.000000e+00, double 3.000000e+00)
  %printf9 = call i8* (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.1, i32 0, i32 0), double %pow32)
  %pow3210 = call double @llvm.pow.f64(double 2.000000e+00, double 1.500000e+00)
  %printf11 = call i8* (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.1, i32 0, i32 0), double %pow3210)
  %powi32 = call double @llvm.powi.f64(double 3.000000e+00, i32 2)
  %printf12 = call i8* (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.1, i32 0, i32 0), double %powi32)
  %powi3213 = call double @llvm.powi.f64(double 5.500000e+00, i32 3)
  %printf14 = call i8* (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.1, i32 0, i32 0), double %powi3213)
  ret i32 0
}

; Function Attrs: nounwind readnone speculatable
declare double @llvm.pow.f64(double, double) #0

; Function Attrs: nounwind readnone speculatable
declare double @llvm.powi.f64(double, i32) #0

attributes #0 = { nounwind readnone speculatable }
