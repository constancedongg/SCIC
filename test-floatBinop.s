	.text
	.file	"scic"
	.section	.rodata.cst8,"aM",@progbits,8
	.p2align	3               # -- Begin function main
.LCPI0_0:
	.quad	4616275186447678439     # double 4.0759999999999996
.LCPI0_1:
	.quad	-4613145184706655945    # double -1.6759999999999999
.LCPI0_2:
	.quad	4614953830317007935     # double 3.4511999999999996
.LCPI0_3:
	.quad	4602678819172646912     # double 0.5
.LCPI0_4:
	.quad	4601193289441697980     # double 0.41753653444676408
.LCPI0_5:
	.quad	4611686018427387904     # double 2
.LCPI0_6:
	.quad	4613937818241073152     # double 3
.LCPI0_7:
	.quad	4609434218613702656     # double 1.5
.LCPI0_8:
	.quad	4621256167635550208     # double 9
.LCPI0_9:
	.quad	4640057816470519808     # double 166.375
	.text
	.globl	main
	.p2align	4, 0x90
	.type	main,@function
main:                                   # @main
	.cfi_startproc
# %bb.0:                                # %entry
	pushq	%r14
	.cfi_def_cfa_offset 16
	pushq	%rbx
	.cfi_def_cfa_offset 24
	pushq	%rax
	.cfi_def_cfa_offset 32
	.cfi_offset %rbx, -24
	.cfi_offset %r14, -16
	leaq	.Lfmt.1(%rip), %r14
	movsd	.LCPI0_0(%rip), %xmm0   # xmm0 = mem[0],zero
	movq	%r14, %rdi
	movb	$1, %al
	callq	printf@PLT
	movsd	.LCPI0_1(%rip), %xmm0   # xmm0 = mem[0],zero
	movq	%r14, %rdi
	movb	$1, %al
	callq	printf@PLT
	movsd	.LCPI0_2(%rip), %xmm0   # xmm0 = mem[0],zero
	movq	%r14, %rdi
	movb	$1, %al
	callq	printf@PLT
	movsd	.LCPI0_3(%rip), %xmm0   # xmm0 = mem[0],zero
	movq	%r14, %rdi
	movb	$1, %al
	callq	printf@PLT
	movsd	.LCPI0_4(%rip), %xmm0   # xmm0 = mem[0],zero
	movq	%r14, %rdi
	movb	$1, %al
	callq	printf@PLT
	leaq	.Lfmt(%rip), %rbx
	movq	%rbx, %rdi
	movl	$1, %esi
	xorl	%eax, %eax
	callq	printf@PLT
	movq	%rbx, %rdi
	movl	$1, %esi
	xorl	%eax, %eax
	callq	printf@PLT
	movq	%rbx, %rdi
	movl	$1, %esi
	xorl	%eax, %eax
	callq	printf@PLT
	movq	%rbx, %rdi
	xorl	%esi, %esi
	xorl	%eax, %eax
	callq	printf@PLT
	movsd	.LCPI0_5(%rip), %xmm0   # xmm0 = mem[0],zero
	movsd	.LCPI0_6(%rip), %xmm1   # xmm1 = mem[0],zero
	callq	pow@PLT
	movq	%r14, %rdi
	movb	$1, %al
	callq	printf@PLT
	movsd	.LCPI0_7(%rip), %xmm1   # xmm1 = mem[0],zero
	movsd	.LCPI0_5(%rip), %xmm0   # xmm0 = mem[0],zero
	callq	pow@PLT
	movq	%r14, %rdi
	movb	$1, %al
	callq	printf@PLT
	movsd	.LCPI0_8(%rip), %xmm0   # xmm0 = mem[0],zero
	movq	%r14, %rdi
	movb	$1, %al
	callq	printf@PLT
	movsd	.LCPI0_9(%rip), %xmm0   # xmm0 = mem[0],zero
	movq	%r14, %rdi
	movb	$1, %al
	callq	printf@PLT
	xorl	%eax, %eax
	addq	$8, %rsp
	.cfi_def_cfa_offset 24
	popq	%rbx
	.cfi_def_cfa_offset 16
	popq	%r14
	.cfi_def_cfa_offset 8
	retq
.Lfunc_end0:
	.size	main, .Lfunc_end0-main
	.cfi_endproc
                                        # -- End function
	.type	.Lfmt,@object           # @fmt
	.section	.rodata.str1.1,"aMS",@progbits,1
.Lfmt:
	.asciz	"%d\n"
	.size	.Lfmt, 4

	.type	.Lfmt.1,@object         # @fmt.1
.Lfmt.1:
	.asciz	"%g\n"
	.size	.Lfmt.1, 4

	.type	.Lfmt.2,@object         # @fmt.2
.Lfmt.2:
	.asciz	"%s\n"
	.size	.Lfmt.2, 4

	.section	".note.GNU-stack","",@progbits
