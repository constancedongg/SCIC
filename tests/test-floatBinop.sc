int func main() {
    printf(1.2 - 0.1);          /* Sub */
    printf(1.2 + 2.876);        /* Add */ 
    printf(1.2 - 2.876);        /* Sub */ 
    printf(1.2 * 2.876);        /* Mul */ 
    printf(1.2 / 2.4);          /* Div */ 
    printf(1.2 / 2.874);        /* Div, truncation at the 6th digit (in total) */
    printb(2.1234567 > 1.5);    /* Greater */
    printb(1.0 == 1.0);         /* Equal - not necessarily correct? */
    printb(1.00 == 1.0);        /* Equal - not necessarily correct? */
    printb(0.99 >= 1.000);      /* Geq */
    printf(2.0 ^ 3.0);         /* Pow float ^ float*/
    printf(2.0 ^ 1.5);         /* Pow float ^ float */
    printf(3.0 ^ 2);         /* Pow float ^ int */
    printf(5.5 ^ 3);            /* Pow float ^ int */
    return 0;
}