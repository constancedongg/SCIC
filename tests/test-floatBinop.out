4.076
-1.676
3.4512
0.5
0.417537
1
1
1
0
8
2.82843
9
166.375
